packages/block-library/src/image/edit.js

```diff
@@ -330,40 +358,49 @@ export function ImageEdit( {
 	);
 	const placeholder = ( content ) => {
 		return (
-			<Placeholder
-				className={ clsx( 'block-editor-media-placeholder', {
-					[ borderProps.className ]:
-						!! borderProps.className && ! isSingleSelected,
-				} ) }
-				withIllustration
-				icon={ lockUrlControls ? pluginsIcon : icon }
-				label={ __( 'Image' ) }
-				instructions={
-					! lockUrlControls &&
-					__(
-						'Upload an image file, pick one from your media library, or add one with a URL.'
-					)
-				}
-				style={ {
-					aspectRatio:
-						! ( width && height ) && aspectRatio
-							? aspectRatio
-							: undefined,
-					width: height && aspectRatio ? '100%' : width,
-					height: width && aspectRatio ? '100%' : height,
-					objectFit: scale,
-					...borderProps.style,
-					...shadowProps.style,
-				} }
-			>
-				{ lockUrlControls ? (
-					<span className="block-bindings-media-placeholder-message">
-						{ lockUrlControlsMessage }
-					</span>
-				) : (
-					content
-				) }
-			</Placeholder>
+			<>
+				{ placeholderResizeListener }
+				<Placeholder
+					className={ clsx( 'block-editor-media-placeholder', {
+						[ borderProps.className ]:
+							!! borderProps.className && ! isSingleSelected,
+					} ) }
+					icon={
+						isLargeContainer &&
+						( lockUrlControls ? pluginsIcon : icon )
+					}
+					withIllustration={
+						! isSingleSelected || ! isLargeContainer
+					}
+					label={ isLargeContainer && __( 'Image' ) }
+					instructions={
+						! lockUrlControls &&
+						isLargeContainer &&
+						__(
+							'Upload an image file, pick one from your media library, or add one with a URL.'
+						)
+					}
+					style={ {
+						aspectRatio:
+							! ( width && height ) && aspectRatio
+								? aspectRatio
+								: undefined,
+						width: height && aspectRatio ? '100%' : width,
+						height: width && aspectRatio ? '100%' : height,
+						objectFit: scale,
+						...borderProps.style,
+						...shadowProps.style,
+					} }
+				>
+					{ lockUrlControls ? (
+						<span className="block-bindings-media-placeholder-message">
+							{ lockUrlControlsMessage }
+						</span>
+					) : (
+						isLargeContainer && content
+					) }
```

Comment:
```suggestion
					{ lockUrlControls && isLargeContainer && (
						<span className="block-bindings-media-placeholder-message">
							{ lockUrlControlsMessage }
						</span>
					) }
					{ ! lockUrlControls && isLargeContainer && content }
```

Not only for `content` but also for `lockUrlControlsMessage `, I think it's better not to render the message itself if the container is not large.

Otherwise, when a small placeholder is selected, the size of the placeholder will change due to the influence of the invisible message:

https://github.com/user-attachments/assets/2aea26b9-ca4b-4949-9502-a0ae12b4f2d4